src/script/interpreter.cpp

```diff
@@ -568,7 +632,7 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                         if (stack.size() < 1)
                             return set_error(serror, SCRIPT_ERR_UNBALANCED_CONDITIONAL);
                         valtype& vch = stacktop(-1);
-                        if (sigversion == SigVersion::WITNESS_V0 && (flags & SCRIPT_VERIFY_MINIMALIF)) {
+                        if (sigversion == SigVersion::TAPSCRIPT || (sigversion == SigVersion::WITNESS_V0 && (flags & SCRIPT_VERIFY_MINIMALIF))) {
```

Comment:
I don't like the idea of changing script validation flags based on context. It might be ok here, but in general that gets complex very quickly - things could even start interacting.

I've just added a comment to explain this. What do you think?

Reply:
It depends on how you define context, does it include `sigversion` parameter ? BIP342 introduction presents well the script rules hierarchy and that would be better if we could encode it in clean code paths, a "flat" flag matrix is obviously harder to reason on.

I think added comment is good enough for now, I guess this conversation belong to the previous, wider one on refactoring/duplicating interpreter before post-taproot script softforks.

nit: `SCRIPT_VERIFY_MINIMALIF` you forgot the `IF` in new comment.

Reply:
> nit: SCRIPT_VERIFY_MINIMALIF you forgot the IF in new comment.

Fixed.